influx-reports/influx.py

```diff
@@ -52,186 +92,232 @@ def status(self, *args, **kwargs):
-    for time_slice in all_data.get_points():
-        for element in time_slice:
-            if element != 'time' and time_slice[element] == 1:
-                key = element.replace('max_', '').replace('_status', '')
-                # Mark the field as up
-                per_field_up_percent[key] += (1 / max_amount_of_time_slices)
-                # Mark the measurement as up
-
-    return dict(
-        per_field_uptime_percent=dict(per_field_up_percent)
-    )
-
-
-def build_report_dict(client, measurements, job_reference,
-                      measurement_period_seconds):
-    report = defaultdict(dict)
-    for measurement in measurements:
-        measurement_data = calculate_measurement_uptime(
-            client, measurement, job_reference, measurement_period_seconds)
-        report['runtime_seconds'] = measurement_period_seconds
-        report['per_field_uptime_percent'][measurement] = \
-            dict((measure, round(measure_uptime * 100, 2)) for
-                 measure, measure_uptime in
-                 measurement_data['per_field_uptime_percent'].items())
-    return dict(report)
-
-
-def main(args=None):
-    client = InfluxDBClient(args.influx_ip, args.influx_port,
-                            database='telegraf')
-    """
-    Influx structure:
-        Measurement (eg maas_nova)
-            Point (a timestamp with data)
-                Tags (Indexed K/V, eg job_reference)
-                Fields (Unindexed K/V, eg keystone_user_count)
-    """
-    # First find the first and last timestamp from telegraf.
-    # This way querying maas_* data will always
-    # be accurate (if no data is reported by maas plugins,
-    # this should be a failure metric)
-    find_first_timestamp_query = (
-        "select first(total) from processes "
-        "where job_reference = '{}';".format(args.job_reference)
-    )
-    first_ts = return_time(client, find_first_timestamp_query, delta_seconds=5)
-    find_last_timestamp_query = (
-        "select last(total) from processes "
-        "where job_reference = '{}';".format(args.job_reference))
-    last_ts = return_time(client, find_last_timestamp_query, delta_seconds=-5)
-
-    measurement_period_seconds = (last_ts - first_ts).total_seconds()
-
-    logging.info(
-        ("Metrics were gathered between {first} and {last}\n"
-         "For a total of {seconds} seconds".format(
-             first=first_ts,
-             last=last_ts,
-             seconds=measurement_period_seconds)))
-
-    measurements = ['maas_glance', 'maas_cinder', 'maas_keystone',
-                    'maas_heat', 'maas_neutron', 'maas_nova', 'maas_horizon']
-
-    report = build_report_dict(
-        client, measurements, args.job_reference, measurement_period_seconds)
-
-    yml_report = yaml.safe_dump(report, default_flow_style=False)
+def generate_reports(data, max_downtime=100, ymlfile=None, subunitfile=None):
+    yml_report = yaml.dump(data, default_flow_style=False)
```

Comment:
This was to workaround something that ended up not being necessary. I'll fix this up.